File: src/Test.CSharp.WinForms/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Nemiro.OAuth;

// For more details, please visit http://oauth.nemiro.net

namespace Test.CSharp.WinForms
{

  public partial class Form2 : Form
  {

    public Form2()
    {
      InitializeComponent();
    }

    public Form2(string providerName)
    {
      InitializeComponent();
      webBrowser1.ScriptErrorsSuppressed = true;
      webBrowser1.Navigate(OAuthWeb.GetAuthorizationUrl(providerName));
    }

    private void webBrowser1_DocumentCompleted(object sender, WebBrowserDocumentCompletedEventArgs e)
    {
      // waiting for results
      if (e.Url.Query.IndexOf("code=") != -1 || e.Url.Fragment.IndexOf("code=") != -1 || e.Url.Query.IndexOf("oauth_verifier=") != -1)
      {
        // is the result, verify
        var result = OAuthWeb.VerifyAuthorization(e.Url.ToString());
        if (result.IsSuccessfully)
        {
          // show user info
          MessageBox.Show
          (
            String.Format
            (
              "User ID: {0}\r\nUsername: {1}\r\nDisplay Name: {2}\r\nE-Mail: {3}", 
              result.UserInfo.UserId,
              result.UserInfo.UserName,
              result.UserInfo.DisplayName ?? result.UserInfo.FullName,
              result.UserInfo.Email
            ), 
            "Successfully", 
            MessageBoxButtons.OK, 
            MessageBoxIcon.Information
          );

          // using access token for API request
          // (this small example for vkontakte only)
          if (result.ProviderName.Equals("vk", StringComparison.OrdinalIgnoreCase))
          {
            new VkontakteFriends(result.AccessTokenValue).ShowDialog();
          }
        }
        else
        {
          // show error message
          MessageBox.Show(result.ErrorInfo.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
        }
        this.Close();
      }
    }

  }

}

File: examples/Test/Test.VB.WinForms/Form2.vb
Imports Nemiro.OAuth

' For more details, please visit http://oauth.nemiro.net

Public Class Form2

  Public Sub New(providerName As String)
    InitializeComponent()
    WebBrowser1.ScriptErrorsSuppressed = True
    WebBrowser1.Navigate(OAuthWeb.GetAuthorizationUrl(providerName))
  End Sub

  Private Sub WebBrowser1_DocumentCompleted(sender As System.Object, e As System.Windows.Forms.WebBrowserDocumentCompletedEventArgs) Handles WebBrowser1.DocumentCompleted
    ' waiting for results
    If Not e.Url.Query.IndexOf("code=") = -1 OrElse Not e.Url.Fragment.IndexOf("code=") = -1 OrElse Not e.Url.Query.IndexOf("oauth_verifier=") = -1 Then
      ' is the result, verify
      Dim result = OAuthWeb.VerifyAuthorization(e.Url.ToString())
      If result.IsSuccessfully Then
        ' show user info
        MessageBox.Show _
        (
          String.Format _
          (
            "User ID: {0}{4}Username: {1}{4}Display Name: {2}{4}E-Mail: {3}",
            result.UserInfo.UserId,
            result.UserInfo.UserName,
            If(Not String.IsNullOrEmpty(result.UserInfo.DisplayName), result.UserInfo.DisplayName, result.UserInfo.FullName),
            result.UserInfo.Email,
            vbNewLine
          ),
          "Successfully",
          MessageBoxButtons.OK,
          MessageBoxIcon.Information
        )

        ' using access token for API request
        ' (this small example for vkontakte only)
        If result.ProviderName.Equals("vk", StringComparison.OrdinalIgnoreCase) Then
          Call New VkontakteFriends(result.AccessTokenValue).ShowDialog()
        End If
      Else
        ' show error message
        MessageBox.Show(result.ErrorInfo.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
      End If
      Me.Close()
    End If
  End Sub

End Class